lib/better_html/parser.rb

```diff
@@ -28,37 +24,40 @@ def initialize(document, template_language: :html)
       else
         raise ArgumentError, "template_language can be :html or :javascript"
       end
-      @nodes = parse!
     end
 
-    private
+    def nodes_with_type(*type)
+      nodes.select{ |node| Array.wrap(type).include?(node.type) }
```

Comment:
Does performance matter here? If so, the `Array.wrap` could be done once up front.